test/built-ins/Temporal/now/timeZone/return-value.js

```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2021 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.now.timezone
+description: Temporal.now.timeZone returns an instance of the TimeZone constructor
+info: |
+  1. Return ? SystemTimeZone().
+features: [Temporal]
+---*/
+
+assert.sameValue(
+  Object.getPrototypeOf(Temporal.now.timeZone()),
+  Temporal.TimeZone.prototype
+);
```

Comment:
This test just tell me that those two things are the same thing, but explicitly that either of them is an instance of `Temporal.TimeZone`. 

`assert(Temporal.now.timeZone() instanceof Temporal.TimeZone)`?

Reply:
The instanceof operator has a lot of additional semantics which would make this test less precise. For example, it would be satisfied if the return value was an instance of a subclass of `Temporal.TimeZone`. I believe a strict equality check is the most precise way to verify this behavior.

Reply:
We could have both?